test/AzExtFsExtra.test.ts

```diff
@@ -0,0 +1,139 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import assert = require('assert');
+import * as fs from 'fs';
+import * as os from 'os';
+import * as path from 'path';
+import { Uri, workspace, WorkspaceFolder } from 'vscode';
+import { AzExtFsExtra, getRandomHexString } from "../extension.bundle";
+import { assertThrowsAsync } from './assertThrowsAsync';
+
+
+suite('AzExtFsExtra', function (this: Mocha.Suite): void {
+    let workspacePath: string;
+    let testFolderPath: string;
+    let workspaceFilePath: string;
+
+    const indexHtml: string = 'index.html';
+
+    const nonExistingPath: string = ' ./path/does/not/exist';
+    const nonExistingFilePath = path.join(nonExistingPath, indexHtml);
+
+    suiteSetup(function (this: Mocha.Context): void {
+        testFolderPath = `azExtFsExtra${getRandomHexString()}`
+        const workspaceFolders: readonly WorkspaceFolder[] | undefined = workspace.workspaceFolders;
+        if (!workspaceFolders) {
+            throw new Error("No workspace is open");
+        }
+
+        workspacePath = workspaceFolders[0].uri.fsPath;
+        workspaceFilePath = path.join(workspacePath, indexHtml);
+    });
+
+    test('pathExists for directory', async () => {
+        assert.strictEqual(await AzExtFsExtra.pathExists(workspacePath), fs.existsSync(workspacePath));
+        assert.strictEqual(await AzExtFsExtra.pathExists(nonExistingPath), fs.existsSync(nonExistingPath));
+    });
+
+    test('pathExists for file', async () => {
+        assert.strictEqual(await AzExtFsExtra.pathExists(workspaceFilePath), fs.existsSync(workspaceFilePath));
+        assert.strictEqual(await AzExtFsExtra.pathExists(nonExistingFilePath), fs.existsSync(nonExistingFilePath));
+    });
+
+    test('isDirectory properly detects folders', async () => {
+        assert.strictEqual(await AzExtFsExtra.isDirectory(workspacePath), isDirectoryFs(workspacePath));
+        assert.strictEqual(await AzExtFsExtra.isDirectory(workspaceFilePath), isDirectoryFs(workspaceFilePath));
+
+    });
+
+    test('isFile properly detects files', async () => {
+        assert.strictEqual(await AzExtFsExtra.isFile(workspaceFilePath), isFileFs(workspaceFilePath));
+        assert.strictEqual(await AzExtFsExtra.isFile(workspacePath), isFileFs(workspacePath));
+    });
+
+    test('ensureDir that does not exist', async () => {
+        const fsPath = path.join(os.homedir(), testFolderPath, getRandomHexString());
+        assert.strictEqual(await AzExtFsExtra.pathExists(fsPath), fs.existsSync(fsPath));
+        await AzExtFsExtra.ensureDir(fsPath);
+
+        assert.strictEqual(await AzExtFsExtra.isDirectory(fsPath), isDirectoryFs(fsPath));
+        assert.strictEqual(await AzExtFsExtra.pathExists(fsPath), fs.existsSync(fsPath));
```

Comment:
This test is for `ensureDir`, so that's the only thing we call from `AzExtFsExtra`. Then we use `fs` to get the "actual" results, and use `true` or `false` as the "expected" result.

```suggestion
        assert.strictEqual(fs.existsSync(fsPath), false);
        await AzExtFsExtra.ensureDir(fsPath);

        assert.strictEqual(isDirectoryFs(fsPath), true);
        assert.strictEqual(fs.existsSync(fsPath), true);
```